paddle/phi/api/yaml/op_compat.yaml

```
@@ -296,6 +296,12 @@
   get_expected_kernel_type :
     assign : GetAssignExpectedKernelType
 
+- op : assign_pos
+  inputs :
+    {x : X, cum_count : cum_count, eff_num_len : eff_num_len}
```

```suggestion
    {x : X}
```
同样的映射就不用注册了，其他地方没什么问题